### ISSUE TITLE
New-UDSelect OnChange Scope Broken

### ISSUE THREAD

**Issue by Viajaz** [ironmansoftware/powershell-universal#1305]
### Steps to Reproduce

`$SelectedOption` isn't set to selected value when `UDSelect` is changed. Is this a scoping problem for the relevant runspace?

See https://forums.ironmansoftware.com/t/udselect-onchange-create-new-content/4456/3 for previously working example.

```powershell
New-UDDashboard -Title "New-UDSelect OnChange Scope Broken" -Content {
    $SelectedOption = 'Unselected'
    New-UDSelect -Option {
        New-UDSelectOption -Name 'One' -Value 1
        New-UDSelectOption -Name 'Two' -Value 2
        New-UDSelectOption -Name 'Three' -Value 3
    } -OnChange {
        $SelectedOption = $EventData
        Show-UDToast -Message "Selection Changed: $EventData"
        Sync-UDElement -Id 'DynSelectedOption'
    }
    New-UDDynamic -Id 'DynSelectedOption' -Content {
        New-UDTypography -Text $SelectedOption
    }
}
```

### Expected behavior

```console
`$SelectedOption` inside of `OnChange` ScriptBlock should update `$SelectedOption` of parent scope.
```

### Actual behavior

```console
`$SelectedOption` inside of `OnChange` ScriptBlock doesn't update `$SelectedOption` of parent scope.
```

### Environment data

PowerShell Universal 3.0.3

**Comment by Viajaz** [comment]
I believe this is happening in the `OnSubmit` scriptblock for `New-UDForm` as well, it has become a showstopper for us halting further development.
I was unable to access global and script scope variables inside of OnSubmit that were exported from an imported Module earlier on in the script.
@adamdriscoll

**Comment by adamdriscoll** [comment]
@Viajaz - This has always been the case and is currently expected behavior. The outer scope is likely running in a different runspace. Variables that we move between runspaces are effectively readonly. Once the runspace is done executing, we reset its state of the runspace so nothing is retained in the local scope. 

You'd have to use session variables to achieve this. 

```
New-UDDashboard -Title "New-UDSelect OnChange Scope Broken" -Content {
    $Session:SelectedOption = 'Unselected'
    New-UDSelect -Option {
        New-UDSelectOption -Name 'One' -Value 1
        New-UDSelectOption -Name 'Two' -Value 2
        New-UDSelectOption -Name 'Three' -Value 3
    } -OnChange {
        $Session:SelectedOption = $EventData
        Show-UDToast -Message "Selection Changed: $EventData"
        Sync-UDElement -Id 'DynSelectedOption'
    }
    New-UDDynamic -Id 'DynSelectedOption' -Content {
        New-UDTypography -Text $Session:SelectedOption
    }
}
```

**Comment by Viajaz** [comment]
@adamdriscoll If I've got a stateful PowerShell Module loaded in the main runspace (Think `Connect-*`, `Disconnect-*` Cmdlets) then I need some way to run commands from said module.

**Comment by Viajaz** [comment]
@adamdriscoll Should I make a new issue for this problem? All my PSU dashboard projects are effectively dead if I can't use my own PowerShell Modules which maintain their own states (Eg: in `$script:` variable scope inside Module scope) between different UD scopes/runspaces.

**Comment by adamdriscoll** [comment]
@Viajaz - I think that's a good idea. It's a fundamental issue with how UD operates so I'm not sure the best solution and might take some discussion to figure out the best way forward.

**Comment by adamdriscoll** [comment]
This issue has been mentioned on **Ironman Software Forums**. There might be relevant details there:

https://forums.ironmansoftware.com/t/using-powershell-modules-and-external-code-in-dashboards/7881/1

**Comment by Viajaz** [comment]
I've created a Forums Thread instead as this issue isn't specific as of yet and requires some discussion, maybe a change in our practice is required.

**Comment by Viajaz** [comment]
> @Viajaz - This has always been the case and is currently expected behavior. The outer scope is likely running in a different runspace. Variables that we move between runspaces are effectively readonly. Once the runspace is done executing, we reset its state of the runspace so nothing is retained in the local scope.

@adamdriscoll Their read-only nature should be mentioned in https://docs.powershelluniversal.com/userinterfaces/dashboards/interaction#variable-scope
